pandas/core/arrays/categorical.py

```diff
@@ -1847,6 +1896,15 @@ def __iter__(self):
         """Returns an Iterator over the values of this Categorical."""
         return iter(self.get_values().tolist())
 
+    def __contains__(self, key):
+        """Returns True if `key` is in this Categorical."""
+        hash(key)
```

Comment:
I think your hash check needs to go into contains itself (null check is ok where it is).

can you add a test on something that isn't hashable as well